src/api/java/AST.java

```diff
@@ -74,14 +63,10 @@ else if (getId() > other.getId())
      * 
      * @return A hash code
      **/
+    @Override
     public int hashCode()
     {
-        int r = 0;
-        try {
-            Native.getAstHash(getContext().nCtx(), getNativeObject());
-        }
-        catch (Z3Exception ex) {}
-        return r;
+        return Native.getAstHash(getContext().nCtx(), getNativeObject());
```

Comment:
By declaration, getAstHash can throw Z3Exception (see Native.java), but hashCode() is not supposed to throw exceptions, so we have to catch it here.